### PR TITLE
Cleanup on gc bridge to support future optimizations

### PR DESCRIPTION
**Pull request by xmcclure**
Consists mostly of
- Comments
- Consolidate duplicate dynarray implementations in new-bridge and tarjan-bridge into a single header full of statics

### CODE REVIEW

File: mono/metadata/sgen-bridge.h [hunk 1]
```diff
@@ -9,20 +9,23 @@
...
  *
- * The output of the SCC analysis is passed to the `cross_references()`
- * callback.  It is expected to set the `is_alive` flag on those strongly
- * connected components that it wishes to be kept alive.  Only bridged objects
- * will be reported to the callback, i.e., non-bridged objects are removed from
- * the callback graph.
+ * The output of the SCC analysis is passed to the client's `cross_references()`
+ * callback.  This consists of 2 arrays, an array of SCCs (MonoGCBridgeSCC),
+ * and an array of "xrefs" (edges between SCCs, MonoGCBridgeXRef).  The edges
+ * are encoded as "API indices", ie indexes in the SCC array.  The client is
```
Comment (kumpera) [inline mono/metadata/sgen-bridge.h]: It's not the edges that are encoded as API indices, but the SCC.

**Comment by kumpera** [comment]
LGTM

File: mono/metadata/sgen-bridge.h (hunk 1 above)
Comment (xmcclure) [inline mono/metadata/sgen-bridge.h]: Would "edges are encoded as pairs of 'API indices'" be a better wording?

**Comment by xmcclure** [comment]
Rephrased the comment kumpera mentioned

**Pull request merged**